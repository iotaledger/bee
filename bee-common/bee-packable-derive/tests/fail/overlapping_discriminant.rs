// Copyright 2021 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(unused_imports, unreachable_patterns)]

use bee_packable::Packable;

#[derive(Packable)]
#[repr(u8)]
#[packable(tag_type = u8)]
pub enum A {
    B = 0,
    #[packable(tag = 0)]
    C,
}

fn main() {}
