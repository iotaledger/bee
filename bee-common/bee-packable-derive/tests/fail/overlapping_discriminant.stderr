error[E0080]: evaluation of constant value failed
 --> tests/fail/overlapping_discriminant.rs:8:10
  |
8 | #[derive(Packable)]
  |          ^^^^^^^^ the evaluated program panicked at 'The tags for the variants `B` and `C` of enum `A` are equal', $DIR/tests/fail/overlapping_discriminant.rs:8:10
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` (in Nightly builds, run with -Z macro-backtrace for more info)
